guide.md

```diff
@@ -0,0 +1,350 @@
+CC targets that were built prior to the addition of this unrelated dependency.
+Instead of carefully managing system images or lists of packages to install
+system-wide in an effectful manner, we use Nix to make them available. The use
+of virtualization or containers is no longer necessary, but we retain the
+reproducibility and predictability benefits of running all builds in identical
+environments.
+
+# Transitioning the build to using Nix
+
+We start by installing Nix as per the official
+[instructions](https://nixos.org/download.html).
+
+```
+> nix --version
+nix (Nix) 2.12.0
+```
+
+rules_nixpkgs requires a Nix-provided Bazel version. We can use the `nix-env
+-i` command to install Bazel into the user environment, but this pollutes the
+shell environment and will prevent the use of the globally installed Bazel (if
+any) in other projects. A better solution is to use `nix-shell`, which can
+create development environments containing specific dependencies on demand,
+somewhat akin to Python virtualenvs but for any kind of dependency. Here's an
+example of how it works:
+
+```
+> hello
+hello: command not found
+> nix-shell -p hello
+nix-shell> hello
+Hello, world!
+nix-shell> exit
+> hello
+hello: command not found
+```
+
+For our purposes, we can enter a shell with a Nix-provided Bazel in a similar manner:
+```
+> nix-shell -p bazel_5
+```
+(`nix-shell` uses `bash` by default; if you use, say, zsh, use `nix-shell -p bazel_5 --run zsh`.)
+
+Once that's done, we can then edit the `WORKSPACE` file to instruct Bazel to load the `rules_nixpkgs` ruleset:
+
+```bazel
+# download the http_archive rule itself
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# download rules_nixpkgs
+http_archive(
+    name = "io_tweag_rules_nixpkgs",
+    strip_prefix = "rules_nixpkgs-9f08fb2322050991dead17c8d10d453650cf92b7",
+    urls = ["https://github.com/tweag/rules_nixpkgs/archive/9f08fb2322050991dead17c8d10d453650cf92b7.tar.gz"],
+    sha256 = "46aa0ca80b77848492aa1564e9201de9ed79588ca1284f8a4f76deb7a0eeccb9",
+)
+
+# load everything that rules_nixpkgs rules need to work
+load("@io_tweag_rules_nixpkgs//nixpkgs:repositories.bzl", "rules_nixpkgs_dependencies")
+rules_nixpkgs_dependencies()
+```
+
+Now we can select a version of the Nixpkgs repository from which all of our
+packages will be drawn. Any Git commit hash or tag is allowed. This way, we
+select a consistent set of package revisions in one go instead of having to
+maintain dependency versions for individual dependencies. Often this will be a
+tag for a stable release, which ensures that there has been especially
+stringent verification that packages in it will function correctly individually
+and play well together. Here we use the 22.05 release:
+
+```bazel
+load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_git_repository")
+nixpkgs_git_repository(
+    name = "nixpkgs",
+    revision = "22.05",
+    sha256 = "",
```

Comment:
Should there be anything about specifying `sha256`, how to find it and why it might be a good idea?

Reply:
Good idea, added.